.pre-commit-config.yaml

```diff
@@ -0,0 +1,43 @@
+repos:
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v3.2.2
+    hooks:
+      - id: pyupgrade
```

Comment:
there are tools here, that we don't use yet.
We should also add those tools to the github actions, without that we can't enforce that someone used the pre-commit. Maybe there's a way to use the pre-commit config in github actions?

Reply:
Either that or I'll copy this from homeassistsnt as well 😀

Reply:
it this works, then we should remove the linting from current github actions
https://pre-commit.ci/